Frends.MicrosoftSQL.ExecuteProcedure/Frends.MicrosoftSQL.ExecuteProcedure/ExecuteProcedure.cs

```diff
@@ -0,0 +1,184 @@
+            {
+                case ExecuteTypes.NonQuery:
+                    dataObject = await command.ExecuteNonQueryAsync(cancellationToken).ConfigureAwait(false);
+                    result = new Result(true, (int)dataObject, null, JToken.FromObject(new { AffectedRows = dataObject }));
+                    break;
+                case ExecuteTypes.Scalar:
+                    dataObject = await command.ExecuteScalarAsync(cancellationToken).ConfigureAwait(false);
+                    result = new Result(true, 1, null, JToken.FromObject(new { Value = dataObject }));
+                    break;
+                case ExecuteTypes.ExecuteReader:
+                    dataReader = await command.ExecuteReaderAsync(cancellationToken).ConfigureAwait(false);
+                    table.Load(dataReader);
+                    result = new Result(true, dataReader.RecordsAffected, null, JToken.FromObject(table));
+                    await dataReader.CloseAsync();
+                    break;
+                default:
+                    throw new NotSupportedException();
+            }
+
+            if (command.Transaction != null)
+                await command.Transaction.CommitAsync(cancellationToken);
+
+            return result;
+        }
+        catch (Exception ex)
+        {
+            if (dataReader != null && !dataReader.IsClosed)
+                await dataReader.CloseAsync();
+
+            if (command.Transaction is null)
+            {
+                var eMsg = $"ExecuteHandler exception: 'Options.SqlTransactionIsolationLevel = None', so there was no transaction rollback. {ex}.";
+                if (options.ThrowErrorOnFailure)
+                    throw new Exception(eMsg);
+                else
+                    return new Result(false, 0, eMsg, null);
+            }
+            else
+            {
+                if (dataReader != null && !dataReader.IsClosed)
+                    await dataReader.CloseAsync();
+
+                try
+                {
+                    await command.Transaction.RollbackAsync(cancellationToken);
+                }
+                catch (Exception rollbackEx)
+                {
+                    var rbEMsg = $"ExecuteHandler exception: An exception occurred on transaction rollback. Rollback exception: {rollbackEx}. ||  Exception leading to rollback: {ex}";
+                    if (options.ThrowErrorOnFailure)
+                        throw new Exception(rbEMsg);
+                    else
+                        return new Result(false, 0, rbEMsg, null);
+                }
+
+                var eMsg = $"ExecuteHandler exception: (If required) transaction rollback completed without exception. {ex}.";
+                if (options.ThrowErrorOnFailure)
+                    throw new Exception(eMsg);
+                else
+                    return new Result(false, 0, eMsg, null);
+            }
+        }
+    }
+
+    private static IsolationLevel GetIsolationLevel(Options options)
```

Comment:
Do not pass the whole `Options` object, you only need its `options.SqlTransactionIsolationLevel` property.